developer-docs-site/docs/standards/aptos-token-v2.md

```diff
@@ -50,10 +50,38 @@ flexibility in this way.
 * NFTs can own other NFTs adding easy composability
 * Soul bound tokens can be easily supported
 
+## Collections and tokens as objects
+In this Token standard, both collections and tokens will be separate [objects](./aptos-object.md). They have their own
+distinct addresses and can be referenced both on and off chain by address. Each object can contain multiple resources
+so collections and tokens are extensible by default, allowing the creator to add custom data and functionalities without
+having to modify the framework.
+
+On chain, another struct can include a reference to the collection or token objects like below:
+```rust
+struct ReferenceExample has key {
+    my_collection: Object<Collection>,
+    my_token: Object<Token>,
+}
+```
+where both `my_collection` and `my_token` are addresses (with `Object<>` wrapper).
+
+Off-chain, the address of an object can be passed along to create a transaction that calls entry functions with `Object<T>`
+as arguments. For example:
+```rust
+public entry fun my_function(my_collection: Object<Collection>) {
+    // Do something with the collection
+}
+```
+
+Collection and token addresses will also be used to query data such as fetching all resources via fullnode API or against
+an indexing service.
+
 ### Royalties
-Royalties are simply [another module](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/royalty.move)
-that is attached to a collection. They're allowed to be updated as long as a `MutatorRef` is generated at creation time.
-The royalty can also be set directly on a token if it has a different royalty config than the collection's.
+Royalties are simply [a resource with associated functionality](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token-objects/sources/royalty.move)
```

Comment:
When you mention "associated functionality", do you mean a "module"?

Reply:
Yep updated to make this clearer